core/src/main/java/org/frankframework/senders/IsolatedServiceCaller.java

```diff
@@ -45,14 +45,14 @@ public class IsolatedServiceCaller {
 	 */
 	@Setter @Getter private TaskExecutor taskExecutor;
 
-	public void callServiceAsynchronous(ServiceClient service, Message message, PipeLineSession session, ThreadLifeCycleEventListener threadLifeCycleEventListener) throws IOException {
-		IsolatedServiceExecutor ise = new IsolatedServiceExecutor(service, message, session, null, threadLifeCycleEventListener);
+	public void callServiceAsynchronous(ServiceClient service, Message message, PipeLineSession session, ThreadLifeCycleEventListener<?> threadLifeCycleEventListener) throws IOException {
+		IsolatedServiceExecutor ise = new IsolatedServiceExecutor(service, message, session, null, threadLifeCycleEventListener, true);
```

Comment:
Kan je niet beter hier een try-with-resource pipelinesession maken?

Reply:
Nee, vanwege de async aanroep in een ander thread is er niet een goed punt om dat te doen, waar je er nog zeker van bent dat de oorspronkelijke session nog open is. En het moet conditioneel wanneer je een async aanroep hebt, niet met een sync. aanroep, dus dat werkt ook niet makkelijk met try-with-resources.

Reply:
Ok, en de callServiceIsolated methode wordt alleen aangeroepen door de (binnenkort gedeprecated IbisLocalSender).

Wat is nu precies het nut van de isolated call aangezien hij de sessie helemaal overneemt? De doc zegt dat het mogelijk iets met transacties doet maar dat is niet zo door de threadconnector. Ik zie dat er geprobeerd is een issue op te lossen door dit toe te voegen maar dat dat niet gelukt is: https://github.com/frankframework/frankframework/issues/4224

Reply:
Iedere sub-adapter heeft sowieso z'n eigen Session, dus is wat dat betreft al geïsoleerd van de hoofd-adapter.

Thread connectors en transacties vind ik voorlopig maar raar en onduidelijk -- Wat Thread Connectors wel nog overbrengen nu is de hide-regexes van de main adapter in het main-thread. Maar je zou kunnen beargumenteren dat een asynchrone / geïsoleerde sub-adapter ook geïsoleerd zou moeten zijn van hide regexes van de aanroeper.

Maar de try-with-resources kan sowieso niet goed vanwege de async aanroep. Je wil de sessie kopieren op het het moment dat je de thread aanmaakt (voordat het thread start), en pas afsluiten op het moment dat de thread klaar is.